### PR TITLE
Feat/find providers

### PR DESCRIPTION
**Pull request by shannonwells**
Closes [Zenhub Retrieval Market Project Issue #12](https://app.zenhub.com/workspaces/retrieval-market-5de558a85ea8f94a5e0882c1/issues/filecoin-project/go-retrieval-market-project/12)

Implement client.FindProviders method which just calls PeerResolver.GetPeers with the desired PieceCID; unit test with test PeerResolver

### CODE REVIEW

**Comment by shannonwells** [comment]
> I have GetPeers taking a CID instead of bytes, because it is a commP...Would you have time to change that as part of this PR? If not it can be done seperately.

Meh, I think it's fine to change it before merging.

File: retrievalmarket/impl/client_test.go [hunk 1]
```diff
@@ -105,10 +109,78 @@ func TestClient_Query(t *testing.T) {
...
+
+	t.Run("when there are no providers", func(t *testing.T) {
+		testResolver := testPeerResolver{peers: []retrievalmarket.RetrievalPeer{}}
+		c := retrievalimpl.NewClient(net, bs, &testRetrievalNode{}, &testResolver)
+		testCid := []byte("unimportant")
+		assert.Len(t, c.FindProviders(testCid), 0)
+	})
+}
+
+type testRetrievalNode struct {
```
Comment (hannahhoward) [inline retrievalmarket/impl/client_test.go]: sorry hate to be a stickeler here, but unless there is a compelling reason, this should use the existing TestRetrievalClientNode -- there's no need to make a dupe here.

Comment (shannonwells) [reply retrievalmarket/impl/client_test.go]: I had this in my branch before you merged yours to master, but agree there should not be two of them.  I overlooked that you had put one in.

Comment (shannonwells) [reply retrievalmarket/impl/client_test.go]: I just missed that it had been moved out of this file when I rebased, is all.

**Pull request merged**